primo/utils/domain_validators.py

```diff
@@ -0,0 +1,44 @@
+#################################################################################
+# PRIMO - The P&A Project Optimizer was produced under the Methane Emissions
+# Reduction Program (MERP) and National Energy Technology Laboratory's (NETL)
+# National Emissions Reduction Initiative (NEMRI).
+#
+# NOTICE. This Software was developed under funding from the U.S. Government
+# and the U.S. Government consequently retains certain rights. As such, the
+# U.S. Government has been granted for itself and others acting on its behalf
+# a paid-up, nonexclusive, irrevocable, worldwide license in the Software to
+# reproduce, distribute copies to the public, prepare derivative works, and
+# perform publicly and display publicly, and to permit others to do so.
+#################################################################################
+
+"""
+This module contains functions that can be used for domain validation
+in Pyomo's `ConfigDict()` data structure.
+"""
+
+
+# pylint: disable-next = invalid-name
```

Comment:
Could you simply reuse in_bounds from primo/util/opt_utils?

Reply:
The `in_bounds` function returns a `bool`. But the domain validator function must return the value if it is in bounds. E.g. If my interval is [5, 10], the `InRange` function returns the value (say 6) if it is the interval, and raises `ValueError` if it is not.